molecule/src/molecule_auto_loader.cpp

```diff
@@ -290,24 +289,58 @@ void MoleculeAutoLoader::_loadMolecule(BaseMolecule& mol, bool query)
     {
         long long pos = _scanner->tell();
         _scanner->skipSpace();
-
         if (_scanner->lookNext() == '{')
         {
-            if (_scanner->findWord("molecule"))
+            if (_scanner->findWord("molecule")) // is it really reliable detection?
             {
+                using namespace rapidjson;
                 _scanner->seek(pos, SEEK_SET);
-                try
+//                try
                 {
-                    MoleculeJsonLoader loader(*_scanner);
-                    if (query)
-                        loader.loadQueryMolecule((QueryMolecule&)mol);
-                    else
-                        loader.loadMolecule((Molecule&)mol);
+                    Array<char> buf;
+                    _scanner->readAll(buf);
+                    buf.push(0);
+                    Document data;
+                    Value rgroups(kArrayType);
+                    Value mol_nodes(kArrayType);
+                    if ( data.Parse(buf.ptr()).HasParseError())
+                      throw Error("Error at parsing JSON: %s", buf.ptr());
```

Comment:
While try/catch is commented, this exception blocks further attempts to recognize the format.

Reply:
The format has already been recognised as ket at this point and if the execution will go further without a throwing an exception it fails somewhere in CML parser which is default for all unrecognised formats so it's better to get terminated here.

Reply:
I can't agree that the format has been recognized at this point. How? by finding `{` and then `"molecule"`? It doesn't look as reliable.
If JSON parser would pass syntax parsing, then OK. But this line is exactly throwing on error after `Parse()` method.

Reply:
Not a lot of chemical formats has "{" and "molecule" then. So in most cases when we get corrupted KET-file "Error at parsing JSON" will appear which is correct behaviour. There are two options how to make the code better:

1) Remove throw Error("Error at parsing JSON: %s", buf.ptr()) to give a try to interpret the data as some other format and let it fail somewhere else which will tangle the error diagnostic.

2) Implement more reliable KET-detection - it makes sense when we'll have some other chemical data format with "{" and "molecule".

Reply:
Extracted to separate https://github.com/epam/Indigo/issues/337